esphome/components/da217/da217.cpp

```diff
@@ -0,0 +1,254 @@
+const uint8_t DA217_ODR_AXIS_DISABLE_Y = 1 << 6;
+const uint8_t DA217_ODR_AXIS_DISABLE_Z = 1 << 5;
+const uint8_t DA217_ODR_AXIS_1HZ = 0b0000;
+const uint8_t DA217_ODR_AXIS_1_95HZ = 0b0001;
+const uint8_t DA217_ODR_AXIS_3_9HZ = 0b0010;
+const uint8_t DA217_ODR_AXIS_7_81HZ = 0b0011;
+const uint8_t DA217_ODR_AXIS_15_63HZ = 0b0100;
+const uint8_t DA217_ODR_AXIS_31_25HZ = 0b0101;
+const uint8_t DA217_ODR_AXIS_62_5HZ = 0b0110;
+const uint8_t DA217_ODR_AXIS_125HZ = 0b0111;
+const uint8_t DA217_ODR_AXIS_250HZ = 0b1000;
+const uint8_t DA217_ODR_AXIS_500HZ = 0b1001;
+
+const uint8_t DA217_REGISTER_MODE_BW = 0x11;
+const uint8_t DA217_MODE_BW_SUSPEND_MODE = 1 << 7;
+const uint8_t DA217_MODE_BW_NORMAL_MODE = 0 << 7;
+
+const uint8_t DA217_REGISTER_INT_SET1 = 0x16;
+const uint8_t DA217_INT_SET1_INT_SOURCE_OVERSAMPLING = 0b00 << 6;
+const uint8_t DA217_INT_SET1_INT_SOURCE_UNFILTERED = 0b01 << 6;
+const uint8_t DA217_INT_SET1_INT_SOURCE_FILTERED = 0b10 << 6;
+const uint8_t DA217_INT_SET1_ENABLE_SINGLE_TAP = 1 << 5;
+const uint8_t DA217_INT_SET1_ENABLE_DOUBLE_TAP = 1 << 4;
+
+const uint8_t DA217_REGISTER_INT_MAP1 = 0x19;
+const uint8_t DA217_INT_MAP1_SM = 1 << 7;
+const uint8_t DA217_INT_MAP1_ORIENT = 1 << 6;
+const uint8_t DA217_INT_MAP1_SINGLE_TAP = 1 << 5;
+const uint8_t DA217_INT_MAP1_DOUBLE_TAP = 1 << 4;
+const uint8_t DA217_INT_MAP1_TILT = 1 << 3;
+const uint8_t DA217_INT_MAP1_ACTIVE = 1 << 2;
+const uint8_t DA217_INT_MAP1_STEP = 1 << 1;
+const uint8_t DA217_INT_MAP1_FREEFALL = 1;
+
+const uint8_t DA217_REGISTER_INT_LATCH = 0x21;
+const uint8_t DA217_INT_LATCH_INT1_LATCH_100MS = 0b1110;
+const uint8_t DA217_INT_LATCH_INT2_LATCH_100MS = 0b1110 << 4;
+
+const uint8_t DA217_REGISTER_TAP_DUR = 0x2A;
+const uint8_t DA217_TAP_DUR_QUIET_30MS = 0b0 << 7;
+const uint8_t DA217_TAP_DUR_QUIET_20MS = 0b1 << 7;
+const uint8_t DA217_TAP_DUR_SHOCK_50MS = 0b0 << 6;
+const uint8_t DA217_TAP_DUR_SHOCK_70MS = 0b1 << 6;
+const uint8_t DA217_TAP_DUR_50MS = 0b000;
+const uint8_t DA217_TAP_DUR_100MS = 0b001;
+const uint8_t DA217_TAP_DUR_150MS = 0b010;
+const uint8_t DA217_TAP_DUR_200MS = 0b011;
+const uint8_t DA217_TAP_DUR_250MS = 0b100;
+const uint8_t DA217_TAP_DUR_370MS = 0b101;
+const uint8_t DA217_TAP_DUR_500MS = 0b110;
+const uint8_t DA217_TAP_DUR_700MS = 0b111;
+
+const uint8_t DA217_REGISTER_TAP_THS = 0x2B;
+const uint8_t DA217_TAP_THS_TAP_TH_MASK = 0b11111;
+
+const float GRAVITY_EARTH = 9.80665f;
+
+void DA217Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up DA217...");
+  uint8_t chip_id;
+  if (!this->read_byte(DA217_REGISTER_CHIP_ID, &chip_id) || (chip_id != 0x13)) {
+    this->mark_failed();
+    return;
+  }
+
+  ESP_LOGV(TAG, "  Setting up bus config...");
+  if (!this->write_byte(DA217_REGISTER_SPI_CONFIG, DA217_SPI_CONFIG_SDO_ACTIVE | DA217_SPI_CONFIG_SOFT_RESET)) {
+    this->mark_failed();
+    return;
+  }
+
+  ESP_LOGV(TAG, "  Setting up resolution...");
+  set_register_(DA217_REGISTER_RESOLUTION_RANGE, resolution_range_);
```

Comment:
Please ensure all member variables and calls are prefixed with `this->`
```suggestion
  this->set_register_(DA217_REGISTER_RESOLUTION_RANGE, resolution_range_);
```